### ISSUE TITLE
Confidential property on Application class

### ISSUE THREAD

**Issue by retroGiant89** [doorkeeper-gem/doorkeeper-mongodb#31]
Mirroring this issue: https://github.com/doorkeeper-gem/doorkeeper/issues/891 . Would it be possible to get the confidential property added to the Doorkeeper:Application class?

**Comment by hiroshi** [comment]
As a workaround, I added those line to config/initializers/doorkeeper.rb
```ruby
Doorkeeper::Application.class_eval do
  field :confidential, type: Boolean, default: true
end
```

It help us from the hell of `NoMethodError (undefined method ``confidential?'` when `POST /oauth/revoke`.

**Comment by retroGiant89** [comment]
I had added the exact same thing in my code. I also ran this:
`Doorkeeper::Application.all.each do |a|
      has_refresh_token = Doorkeeper::AccessToken.where(application_id: a.id, :refresh_token.exists => true).present?
      a.update(confidential: has_refresh_token)
      puts "Setting Oauth App '#{a.name}' confidential to #{has_refresh_token}"
    end`

To set the confidential field appropriately.

It would still be better if this was dealt with in the gem tho

**Comment by nbulaj** [comment]
Related to #32

**Comment by nbulaj** [comment]
Released with 4.2.0 version, available through rubygems

**Comment by falak** [comment]
add a boolean field in oauth_applications existing table in your rails app will do the trick for you.
`add_column :oauth_applications, :confidential, :boolean, default: true, null: false`
